apps/dashboard/app/javascript/dynamic_forms.js

```diff
@@ -670,13 +670,13 @@ function optionForFromToken(str) {
     };
 
     if(hide) {
-      option.style.display = 'none';
-      option.disabled = true;
-
       if(option.selected) {
         option.selected = false;
         hideSelectedValue = option.textContent;
       }
+
+      option.style.display = 'none';
+      option.disabled = true;
```

Comment:
I think I had to make this change because it was working by coincidence.

This PR has duplicate options - i.e., options with the same `value` but different `data-` directives which we've gone back and forth on, but yea I suspect it was working by coincidence before because there were fewer (or no) duplicate options.